### PR TITLE
Make GatewayApi async

### PR DESCRIPTION
**Pull request by BrianMaki**
# Implements [AB#16544](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/16544)

## Description

Made GatewayApi async and added Cancellation Token to:

- Controllers
- Services: Added DB result checking in some cases as that DB result was never checked and assumed to be successful
- Delegates
- Updated unit tests

Created [AB#16590](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/16590) to convert Job Scheduler to async for delegates calls that were not converted to async in this ticket.

Also created [AB#16591](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/16591) to veirify and refactor unit tests:

- Clean up and simplify unit tests
- Some of the mocks are no longer due to previous code changes (i.e., not this PR)

Screenshot 2023-12-18 at 3 00 06 PM

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## UI Changes

None

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Comment by BronzBierd** [comment]
There are some of the synchronous methods that are currently just maintaining the jobScheduler and the Tests.

I've update the description of the AB#16591 to include removing Synchronous methods that are no longer used after the tests have been refactored.

**Review by BronzBierd — APPROVED** [review]
Its a little difficult to follow but looking good I see where you want to take this. I believe when you bring the unit tests in line it will give a good indication of the real change.

**Review by sslaws — APPROVED** [review]

File: Apps/Common/src/Services/EmailQueueService.cs [hunk 1]
```diff
@@ -64,18 +68,39 @@ public void QueueNewEmail(string toEmail, string templateName, bool shouldCommit
             this.QueueNewEmail(toEmail, templateName, keyValues, shouldCommit);
         }
 
+        /// <inheritdoc/>
+        public async Task QueueNewEmailAsync(string toEmail, string templateName, bool shouldCommit = true, CancellationToken ct = default)
+        {
+            Dictionary<string, string> keyValues = new();
+            await this.QueueNewEmailAsync(toEmail, templateName, keyValues, shouldCommit, ct);
+        }
```
[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/Common/src/Services/EmailQueueService.cs:76]: That line does not exist anymore in the latest changes committed.

[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/Common/src/Services/EmailQueueService.cs:76]: Used in UserProfileService.

Screenshot 2023-12-20 at 11 55 49 AM

[the remark this answers is not in the document]
Comment (BrianMaki) [reply Apps/Common/src/Services/EmailQueueService.cs:76]: I will review again in AB#116590

**Pull request merged**